chronicles/textblockserializer.nim

```diff
@@ -0,0 +1,113 @@
+import
+  times,
+  strutils,
+  terminal,
+  system,
+  options
+
+import
+  serialization,
+  faststreams/[outputs, textio]
+
+import
+  chronicles,
+  options
+
+type
+  TextBlockWriter*[timeFormat: static[TimestampsScheme], colorScheme: static[ColorScheme]] = object
+    stream: OutputStream
+    currentLevel: LogLevel
+
+  TextBlockReader* = object
+    lexer: string
+
+serializationFormat TextBlockLog,
+                    Reader = TextBlockReader,
+                    Writer = TextBlockWriter,
+                    PreferedOutput = string,
+                    mimeType = "text/plain"
+
+#
+# Class startup
+#
+
+proc init*(w: var TextBlockWriter, stream: OutputStream) =
+  w.stream = stream
+
+#
+# Field Handling
+#
+
+proc writeFieldName*(w: var TextBlockWriter, name: string) =
+  w.stream.writeText name
+  w.stream.writeText ": "
+
+proc writeValue*(w: var TextBlockWriter, value: auto) =
+  w.stream.writeText value
```

Comment:
You need code here that will call the right `write` routine. For example:

```nim
when value is array:
  writeIterable(w, value)
elif value is object:
  w.stream.write '{'
  enumInstanceSerializedFields(value, fieldName, filed):
    writeField(fieldName, field)
  w.stream.write '}'
else ...
```

Reply:
I've now made changes of that nature in both textBlock and textLine.

Since this changes the output of textBlock, I went ahead and enabled growing tabs recursively for objects within objects within object etc.